protocol/interoperable-ether.md

```diff
@@ -0,0 +1,198 @@
+to make the migration extra simple.
+
+An important problem to solve is ensuring that there are no liquidity constraints. If a wrap/unwrap mechanism
+is used rather than a mint/burn mechanism, it will result in liquidity constraints. Therefore we prefer a 
+mint/burn mechanism. We need a solution to the liquidity constraint that specifically happens when a user is
+trying to send ether to a remote domain. They have to wrap the ether into weth and then it gets sent between
+chains as weth and then we need to unwrap the weth. Its possible that there isn’t enough weth to unwrap on the other side.
+
+We introduce 2 new predeploys
+
+- `SuperchainWETH`
+    - Implements `SuperchainERC20`
+    - Guaranteed to be wrapped ether rather than wrapped native asset
+- `ETHLiquidity`
+    - Contains an artificially large pool of `ether` to provide liquidity for cross chain sends
+    - We “burn” ether by sending it here and “mint” ether by pulling it out
+    - Only `SuperchainWETH` can interact with this contract
+    - Only works on non custom gas token chains
+    - This is similar to the Scroll bridge where in genesis they mint a ton of ether into the contract where it can only be unlocked via a L1 to L2 deposit
+    - Placed in genesis with a balance equal to `type(uint248).max`
+
+The important invariant is that `ether` cannot be withdrawn from the `SuperchainWETH` contract by entities that do not own it.
+This would result in inflating the supply of ether.
+
+### SuperchainWETH
+
+- Users can send ether between chains as this contract will handle the wrapping and unwrapping, this only works when
+  the source and destination are not custom gas token
+- Users can send ether from a non custom gas token chain to a custom gas token chain and it will end up as weth
+- Users can send weth from a custom gas token chain to a non custom gas token chain and it will end up as ether
+- Users can send weth between custom gas token chains
+
+```solidity
+contract SuperchainWETH is WETH98, ISuperchainERC20 {
+  L1Block internal l1Block = L1Block(Predeploys.L1_BLOCK);
+  L2ToL2CrossDomainMessenger messenger = L2ToL2CrossDomainMessenger(Predeploys.L2_TO_L2_CROSSDOMAIN_MESSENGER);
+  ETHLiquidity liquidity = ETHLiquidity(Predeploys.ETH_LIQUIDITY);
+
+  function deposit() public payable override {
+    if (l1block.isCustomGasToken()) revert IsCustomGasToken();
+    super.deposit();
+  }
+
+  function withdraw(uint256 wad) public override {
+    if (l1block.isCustomGasToken()) revert IsCustomGasToken();
+    super.withdraw(wad);
+  }
+
+  function sendETHTo(address to, uint256 chainId) public payable {
+    if (l1block.isCustomGasToken()) revert IsCustomGasToken();
+
+    liquidity.burn{ value: msg.value }();
+
+    sendMessage({
+      _destination: chainId,
+      _message: abi.encodeCall(SuperchainWETH.relayETH, (to, msg.value))
+    });
+  }
+
+  function sendERC20(uint256 wad, uint256 chainId) {
```

Comment:
From the user perspective, we are calling a WETH contract to move Ether (either natively, or in wrapped form between chains). Therefore I think it would be nice if we can simplify the interface to remove unify the `sendETHTo` and `sendERC20To` method into a single `sendWETHTo` method. Similar for `relayETH` vs. `relayERC20`. This method would be have conditionals of course

```solidity
function sendWETHTo(address to, uint256 wad, uint256 chainId) external payable {
  if (l1block.isCustomGasToken()) {
    require(msg.value == 0);
  } else {
    require(msg.value == wad);
  }
}
```

I'd also be ok with adding an overload that has no `wad` input for chains where the gas token is ETH. But having the (W)ETH holder worry about the chain's gas token with different named methods feels confusing

Reply:
I think this could certainly simplify things, this is a good suggestion. Ideally there is a smaller interface